platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -77,15 +78,8 @@ void OfflineDatabase::changePath(const std::string& path_) {
 }
 
 void OfflineDatabase::cleanup() {
-    // Deleting these SQLite objects may result in exceptions
-    try {
-        statements.clear();
-        db.reset();
-    } catch (const util::IOException& ex) {
-        handleError(ex, "close database");
-    } catch (const mapbox::sqlite::Exception& ex) {
-        handleError(ex, "close database");
-    }
+    statements.clear();
+    db.reset();
```

Comment:
we should fix throwing destructor from QT port https://github.com/mapbox/mapbox-gl-native/blob/master/platform/qt/src/sqlite3.cpp#L57

Reply:
@alexshalamov This PR now includes only changes that are relevant to the new API. Let's handle re-factoring DB code in a separate PR.